ad-joining/register-computer/main.py

```diff
@@ -344,23 +322,56 @@ def __register_computer(request):
         else:
             computer_name = auth_info.get_instance_name()
 
-        logging.info("Successfully read GCE instance data for '%s' (hostname: '%s'), authorized (2/2)" %
+        logging.info("Successfully read GCE instance data for '%s' (hostname: '%s'), authorized (1/2)" %
             (auth_info.get_instance_name(), computer_name))
     except Exception as e:
         logging.exception("Checking project access to '%s' failed" % auth_info.get_project_id())
         return flask.abort(HTTP_ACCESS_DENIED)
-    
-    # If custom OU is being used, then extract it from the compute instance 
-    if "CUSTOM_OU_ROOT_DN" in os.environ:
+
+    # Authorize, Part 2: Check that there is a OU with the same name as the
+    # project id. The existence of such a OU implies that the owner of the
+    # domain is OK with joining machines from that project.
+    # This check is ignored if the service is configured to use a custom OU.
+    # For custom OU, users will need to use an external way to make sure rogue/unauthorized
+    # projects cannot access the service, for example, by using a Shared VPC with Service projects.
+    computer_ou = None
+    if "PROJECTS_DN" in os.environ and "CUSTOM_OU_ROOT_DN" in os.environ:
```

Comment:
maybe move this logic to a separate method?

Reply:
Keeping it under consideration. Will apply on future fix.